test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SqlProjects/SqlProjectsServiceTests.cs

```diff
@@ -189,6 +189,99 @@ await service.HandleDeleteSqlObjectScriptRequest(new SqlProjectScriptParams()
             Assert.IsFalse(File.Exists(movedScriptAbsolutePath), $"{movedScriptAbsolutePath} expected to have been deleted from disk");
         }
 
+        [Test]
+        public async Task TestNoneScriptOperations()
+        {
+            // Setup
+            SqlProjectsService service = new();
+            string projectUri = await service.CreateSqlProject();
+            Assert.AreEqual(0, service.Projects[projectUri].NoneScripts.Count, "Baseline number of NoneScripts");
+
+            // Validate adding a SQL object script
+            MockRequest<ResultStatus> requestMock = new();
+            string relativePath = "NoneIncludeFile.json";
+            string absolutePath = Path.Join(Path.GetDirectoryName(projectUri), relativePath);
+            
+            #pragma warning disable JSON002 // Probable JSON string detected
+            await File.WriteAllTextAsync(absolutePath, @"{""included"" : false }");
+            #pragma warning restore JSON002 // Probable JSON string detected
+            
+            Assert.IsTrue(File.Exists(absolutePath), $"{absolutePath} expected to be on disk");
+
+            await service.HandleAddNoneScriptRequest(new SqlProjectScriptParams()
+            {
+                ProjectUri = projectUri,
+                Path = relativePath
+            }, requestMock.Object);
+
+            requestMock.AssertSuccess(nameof(service.HandleAddNoneScriptRequest));
+            Assert.AreEqual(1, service.Projects[projectUri].NoneScripts.Count, "NoneScripts count after add");
+            Assert.IsTrue(service.Projects[projectUri].NoneScripts.Contains(relativePath), $"NoneScripts expected to contain {relativePath}");
+
+            // Validate getting a list of the SQL object scripts
+            MockRequest<GetScriptsResult> getMock = new();
+            await service.HandleGetNoneScriptsRequest(new SqlProjectParams()
+            {
+                ProjectUri = projectUri
+            }, getMock.Object);
+
+            getMock.AssertSuccess(nameof(service.HandleGetNoneScriptsRequest));
+            Assert.AreEqual(1, getMock.Result.Scripts.Length);
```

Comment:
one more thing to test is that when a project already has a pre/post deploy script, that trying to add another pre/post deploy script gets added as a none script

Reply:
Shouldn't that be tested down in the DacFx layer since that's where the relevant code is?  Though I'm not spotting that particular case in the test; if I'm not completely overlooking it, I'll add it there.

https://msdata.visualstudio.com/SQLToolsAndLibraries/_git/DacFx?path=/Test/UnitTests/UTProjects/NonBuildSqlScriptTests.cs

Reply:
ah yeah it's already tested here https://msdata.visualstudio.com/SQLToolsAndLibraries/_git/DacFx?path=/Test/UnitTests/UTProjects/NonBuildSqlScriptTests.cs